readthedocs/doc_builder/environments.py

```diff
@@ -275,15 +276,17 @@ class BuildEnvironment(object):
     :param build: Build instance
     :param record: Record status of build object
     :param environment: shell environment variables
+    :param finalize: finalize the build by setting a finished state on exit
```

Comment:
this sentence is `True` _only_ if the `update_build` method is called with the `BUILD_STATE_FINISHED` status, otherwise that method will just update the build, but not finalize it

Reply:
Ah true. I didn't like `finalize` either. I'll rename this `commit` to match the Django usage. Would this change make more sense?

Reply:
maybe the `commit` should be in the `update_build` method to avoid the other problems that I mentioned